### DISCUSSION TITLE
Usage for really big and complex systems

### DISCUSSION THREAD

**Discussion by almorgv** [structurizr/java#290] [Q&A]
### Question

Hi! We are researching how to use Structurizr for a very complex system with dozens of subsystems and hundreds of microservices.
Do you have any suggestions on how this might be achieved?
Does Structurizr support dynamic DSL rendering to handle navigation between subsystems? 
So can we do something like a tree structure with includes but on a scale of hundreds (maybe thousands) of services?
```
.
│   workspace.dsl
│
└───subsystem-1
│   └───service-11
│       │   service.dsl
│   
│   └───service-12
│       │   service.dsl
│   
└───subsystem-2
│   └───service-21
│       │   service.dsl
│   
│   └───service-22
│       │   service.dsl
```

**Comment by simonbrowndotje** [comment]
My first recommendation would be to ensure that you are using C4 correctly - a single software system with hundreds of services sounds a little strange to me. My second would be to look at the enterprise examples for some inspiration -> https://github.com/structurizr/examples/tree/main/enterprise#example-1
